### DISCUSSION TITLE
Thoughts on exposing uploadTraces/uploadMetrics?

### DISCUSSION THREAD

**Discussion by codeboten** [open-telemetry/opentelemetry-go#1867] [General]
I was trying to translate metrics from Telegraf using the Otel Go SDK (https://github.com/influxdata/telegraf/pull/9109/files) and found myself re-using the grpc client in the opentelemetry-prometheus-sidecar (https://github.com/lightstep/opentelemetry-prometheus-sidecar/blob/main/otlp/client.go) instead of using the protocol driver.

I would be able to re-use the otlp grpc client code if uploadMetrics was available, as I already have a slice of `ResourceMetrics`. My question is whether there's a better way to go about this? Or if it would be possible to make that function available?

Either ways, I completely understand the API/SDK will change shortly so I guess what I'm looking for here is guidance from the community and wanted to provide this feedback.

**Comment by Aneurysm9** [comment]
This is definitely something that would be nice to enable.  I think that the work @paivagustavo is doing on https://github.com/open-telemetry/opentelemetry-go/issues/1827 may enable this.  Can you comment on that issue if you have any comments or concerns from this perspective?
